source/common/common/mem_block_builder.h

```diff
@@ -0,0 +1,126 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "common/common/assert.h"
+
+namespace Envoy {
+
+// Manages a block of raw memory for objects of type T. T must be
+// empty-constructible. This class carries extra member variables for tracking
```

Comment:
Do you mean `DefaultConstructible` here? There isn't a named requirements called empty-constructible.

https://en.cppreference.com/w/cpp/named_req/DefaultConstructible